src/sage/rings/function_field/function_field.py

```diff
@@ -1224,6 +1227,131 @@ def completion(self, place, name=None, prec=None, gen_name=None):
+            Place (1/x)
+            sage: a = (5*x + 6)/(x + 15)
+            sage: b = 7/x
+            sage: K.hilbert_symbol(a, b, P)
+            -1
+
+            sage: Q = K.places()[7]; Q
+            Place (x + 6)
+            sage: c = 15*x + 12
+            sage: d = 16/(x + 13)
+            sage: K.hilbert_symbol(c, d, Q)
+            1
+
+            sage: K.<x> = FunctionField(GF(5)); R.<T> = PolynomialRing(K)
+            sage: f = ((x^2 + 2*x + 2)*T^5 + (4*x^2 + 2*x + 3)*T^4 + 3*T^3 + 4*T^2
+            ....:     + (2/(x^2 + 4*x + 1))*T + 3*x^2 + 2*x + 4)
+            sage: L.<y> = K.extension(f)
+            sage: P = L.places_above(K.places()[0])[1]
+            sage: a = (((3*x^3 + 2*x^2 + x + 1)/(x + 2))*y^4 + (4/(x + 4))*y^3
+            ....:     + ((3*x + 4)/(x + 4))*y^2
+            ....:     + ((x^4 + 2*x^3 + 3*x^2 + x + 2)/(x^4 + x^3 + x^2 + 2))*y
+            ....:     + (x^4 + x^2 + 2*x + 4)/(x^2 + 2*x + 2))
+            sage: b = (((x + 1)/(x + 4))*y^4
+            ....:     + ((2*x^4 + 3*x^3 + 3*x^2 + 3)/(x^2 + 2*x + 2))*y^3 + (2/(x + 4))*y^2
+            ....:     + ((4*x^4 + 4*x^3 + 4*x^2 + 2)/(x^4 + x^3 + x^2 + 2))*y
+            ....:     + (x^3 + x)/(x^3 + x^2 + x + 2))
+            sage: L.hilbert_symbol(a, b, P)
+            1
+            sage: Q = L.places_above(K.places()[1])[0]
+            sage: c = (((3*x^3 + 3*x^2 + 3*x + 3)/(x^3 + 3*x^2 + x))*y^4
+            ....:     + ((3*x^4 + 2*x^3 + 2*x^2 + 4*x + 3)/(x^3 + 3*x^2 + x))*y^3
+            ....:     + ((3*x + 4)/(x^3 + 3*x^2 + x))*y^2
+            ....:     + ((4*x + 2)/(x^5 + 2*x^4 + 4*x^3 + 2*x^2 + x))*y
+            ....:     + (x^3 + 2*x^2 + 4)/(x^3 + 3*x^2 + x))
+            sage: d = (((2*x^4 + x^2 + 2*x + 3)/(x^2 + 2*x + 2))*y^4
+            ....:     + ((4*x^2 + 2)/(x^2 + 2*x + 2))*y^3 + ((3*x^2 + 3*x + 3)/(x^2 + 1))*y^2
+            ....:     + ((x^6 + 4*x^5 + 4*x^4 + 3*x^3 + 3*x^2 + x + 3)/(x^4 + x^3 + x^2 + 2))*y
+            ....:     + (4*x^3 + 4*x^2 + 2*x + 2)/(x + 4))
+            sage: L.hilbert_symbol(c, d, Q)
+            -1
+
+            sage: K.<x> = FunctionField(GF(3)); R.<T> = PolynomialRing(K)
+            sage: g = (((2*x + 1)/x)*T^5 + ((2*x + 1)/(x + 1))*T^4 + ((x^2 + 1)/x)*T^3
+            ....:     + (2*x/(x^2 + 2*x + 2))*T^2 + 2*T + (2*x + 2)/(x + 2))
+            sage: L.<y> = K.extension(g)
+            sage: P = L.places_above(K.places()[1])[1]
+            sage: a = (((x + 2)/(x + 1))*y^4 + ((x^4 + 2*x^2 + 1)/(x^2 + 2*x))*y^3
+            ....:     + ((x^5 + x^2 + 2*x + 1)/(x^4 + 2*x^3 + x^2 + x + 1))*y^2
+            ....:     + ((2*x + 2)/(x + 2))*y + (2*x^4 + x^3 + x^2 + 2*x + 2)/(x^2 + x + 1))
+            sage: b = (((x^2 + x + 1)/(x^2 + 2*x + 1))*y^4 + ((2*x^2 + x + 1)/(x + 1))*y^3
+            ....:     + ((x^5 + x^4 + 2*x^3 + x^2 + 2*x + 2)/(x^3 + x + 2))*y^2
+            ....:     + (2*x/(x^2 + 2))*y + 2*x^2 + 2*x + 1)
+            sage: L.hilbert_symbol(a, b, P)
+            -1
+        """
```

Comment:
Can you use simpler functions for a, b, c, d? Ideally that can be input in one line.

It would be good to make up an example to demonstrate an arithmetic property of the Hilbert symbol, like the bimultiplicative property.

Reply:
Thanks for the idea! This actually helped me discover a bug because the application of the residue field map doesn't work as I expected - I fixed it and included random examples for bimultiplicativity.